src/portfft/common/workgroup.hpp

```diff
@@ -53,6 +58,45 @@ constexpr T bank_lines_per_pad_wg(T row_size) {
 }
 
 namespace detail {
+
+// struct for the result of factorize_for_wg
+struct wg_factorization {
+  Idx factor_wi_n;
+  Idx factor_sg_n;
+  Idx factor_wi_m;
+  Idx factor_sg_m;
+};
+
+/** Calculate a valid factorization for workgroup dfts, assuming there is sufficient local memory.
+ * @tparam Scalar scalar type of the transform data
+ * @param fft_size the number of elements in the transforms
+ * @param subgroup_size the size of subgroup used for the transform
+ * @return a factorization for workgroup dft or null if the size won't work with the implemenation of workgroup dfts.
+ */
+template <typename Scalar>
+std::optional<wg_factorization> factorize_for_wg(IdxGlobal fft_size, Idx subgroup_size) {
```

Comment:
Lets move this to `utils.hpp`, we have only device callable functions in the common folder.

Reply:
Currently we have the factorization functions for workitem and subgroup in `common/workitem.hpp` and `common/subgroup.hpp` respectively, so I was following the example there.
`factorize_sg` is not called from device anywhere, along with `fits_in_sg` and `fits_in_wi`, so I wouldn't say we only have device callable functions in the common folder.
If we do want to refactor to puts the factorization functions in a utility file, then we should group them and put them in a "factorization.hpp" or something like that. Generic util files are a bit of a code smell imo (though I am guilty of committing that sin).

Reply:
I'd agree that a `factorisation.hpp` would be better than having everything in `utils.hpp`.